pkg/reconciler/apis/apiexport/apiexport_controller_test.go

```diff
@@ -0,0 +1,245 @@
+/*
+Copyright 2022 The KCP Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apiexport
+
+import (
+	"context"
+	"crypto/sha256"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/kcp-dev/apimachinery/pkg/logicalcluster"
+	"github.com/stretchr/testify/require"
+
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	apisv1alpha1 "github.com/kcp-dev/kcp/pkg/apis/apis/v1alpha1"
+	conditionsv1alpha1 "github.com/kcp-dev/kcp/third_party/conditions/apis/conditions/v1alpha1"
+	"github.com/kcp-dev/kcp/third_party/conditions/util/conditions"
+)
+
+func TestReconcile(t *testing.T) {
+	tests := map[string]struct {
+		secretRefSet                         bool
+		secretExists                         bool
+		createSecretError                    error
+		wantGenerationFailed                 bool
+		keyMissing                           bool
+		secretHashDoesntMatchAPIExportStatus bool
+		apiExportHasExpectedHash             bool
+		apiExportHasSomeOtherHash            bool
+		hasPreexistingVerifyFailure          bool
+		wantError                            bool
+		wantCreateSecretCalled               bool
+		wantUnsetIdentity                    bool
+		wantDefaultSecretRef                 bool
+		wantStatusHashSet                    bool
+		wantVerifyFailure                    bool
+		wantIdentityValid                    bool
```

Comment:
taste comment: would add some empty lines between block of similar fields. Makes it easier to read.

Reply:
same below in the test cases, at least <input><empty line><expected>

Reply:
Fixed